contrib/vagrant/control-plane-base.sh

```diff
@@ -1,6 +1,161 @@
-systemctl restart crio
-systemctl enable kubelet.service
+# # zypper install -y -t pattern kubic_admin
+# zypper install -y cri-o
+# # # install -D -m 0644 /home/vagrant/enclave/contrib/crio/00-default.conf /etc/crio/crio.conf.d/00-default.conf
+# mv /etc/crio/crio.conf.d/00-default.conf /etc/crio/crio.conf
+# systemctl enable crio
+# systemctl start crio
+# # systemctl restart crio
+# # systemctl enable kubelet.service
+
+cat <<EOF > /etc/containerd/config.toml
+root = "/var/lib/containerd"
+state = "/run/containerd"
+oom_score = 0
+
+[grpc]
+  address = "/run/containerd/containerd.sock"
+  uid = 0
+  gid = 0
+  max_recv_message_size = 16777216
+  max_send_message_size = 16777216
+
+[debug]
+  address = ""
+  uid = 0
+  gid = 0
+  level = ""
+
+[metrics]
+  address = ""
+  grpc_histogram = false
+
+[cgroup]
+  path = ""
+
+[plugins]
+  [plugins.cgroups]
+    no_prometheus = false
+  [plugins.cri]
+    stream_server_address = ""
+    stream_server_port = "10010"
+    enable_selinux = false
+    sandbox_image = "k8s.gcr.io/pause:3.2"
+    stats_collect_period = 10
+    systemd_cgroup = true
+    enable_tls_streaming = false
+    max_container_log_line_size = 16384
+    [plugins.cri.containerd]
+      snapshotter = "overlayfs"
+      no_pivot = true
+      [plugins.cri.containerd.default_runtime]
+        runtime_type = "io.containerd.runtime.v1.linux"
+        runtime_engine = ""
+        runtime_root = ""
+      [plugins.cri.containerd.untrusted_workload_runtime]
+        runtime_type = ""
+        runtime_engine = ""
+        runtime_root = ""
+    [plugins.cri.cni]
+      bin_dir = "/opt/cni/bin"
+      conf_dir = "/etc/cni/net.d"
+      conf_template = ""
+    [plugins.cri.registry]
+      [plugins.cri.registry.mirrors]
+        [plugins.cri.registry.mirrors."docker.io"]
+          endpoint = ["https://registry-1.docker.io"]
+        [plugins.diff-service]
+    default = ["walking"]
+  [plugins.linux]
+    shim = "containerd-shim"
+    runtime = "/home/vagrant/.cargo/bin/enclave-runc-wrapper"
+    runtime_root = ""
+    no_shim = false
+    shim_debug = false
+  [plugins.scheduler]
+    pause_threshold = 0.02
+    deletion_threshold = 0
+    mutation_threshold = 100
+    schedule_delay = "0s"
+    startup_delay = "100ms"
+EOF
+
+cat <<EOF > /etc/docker/daemon.json
+{
+  "log-level": "warn",
+  "log-driver": "json-file",
+  "log-opts": {
+    "max-size": "10m",
+    "max-file": "5"
+  },
+  "default-runtime": "runc-enclave",
+  "runtimes": {
+    "runc-enclave": {
+      "path": "/home/vagrant/.cargo/bin/enclave-runc-wrapper"
+    }
+  }
+}
+EOF
+
+cat <<EOF > /etc/systemd/system/containerd.service
+[Unit]
+Description=containerd container runtime
+Documentation=https://containerd.io
+After=network.target local-fs.target
+
+[Service]
+ExecStartPre=-/sbin/modprobe overlay
+ExecStartPre=/usr/bin/mkdir -p /sys/fs/cgroup/systemd
+ExecStartPre=/usr/bin/mount -t cgroup -o none,name=systemd cgroup /sys/fs/cgroup/systemd
```

Comment:
Permissions aren't an issue here?

Reply:
Apparently not - containerd works fine after this mount.

After ensuring that this unit works for longer time, I'm going to submit this unit file to openSUSE containerd package - currently it's not there.

Reply:
Sounds good!